tests/conftest.py

```diff
@@ -42,14 +54,19 @@ def get_instance(cls):
     def __init__(self):
         self._tmpdir = tempfile.mkdtemp()
         self._port = 27017
-        self._process = subprocess.Popen(['mongod', '--bind_ip', 'localhost',
-                                          '--port', str(self._port),
-                                          '--dbpath', self._tmpdir,
-                                          '--nojournal',
-                                          '--noauth',
-                                          '--syncdelay', '0'],
-                                         stdout=open('/tmp/mongo-temp.log', 'wb'),
-                                         stderr=subprocess.STDOUT)
+        
+        try:
+            self._process = subprocess.Popen(['mongod', '--bind_ip', 'localhost',
+                                              '--port', str(self._port),
+                                              '--dbpath', self._tmpdir,
+                                              '--nojournal',
+                                              '--noauth',
+                                              '--syncdelay', '0'],
+                                             stdout=open('/tmp/mongo-temp.log', 'wb'),
+                                             stderr=subprocess.STDOUT)
+        except FileNotFoundError as e:
+            logger.warning("Mongodb executable not found, trying to use docker compose mongodb")           
```

Comment:
I am confused by this message - how is this related to docker-compose?

Reply:
If mongodb Is not installed It Will be used the One executed in the docker compose

I found a misalignment between the readme and the code, regarding the execution of the tests

We can improve, this Is Just a step forward or, better, just a fix

Reply:
Sorry, still at a loss, maybe due to the wording.

Does the message say that it failed to run mongodb _inside docker compose_ - or it failed to find mongodb in the default path and then it will try via docker compose?

Reply:
the second,

it falied the run of mongod locally, and it assumes that you've follow the instructions in the readme and have correctly executed the docker compose.

We can improve here, so consider me open for a proposal

Reply:
Thanks for bearing with me.

I now understand the logic in that code.

It was: "always run mongodb as a sub-process".

It now is:
* try running mongodb as a sub-process.
* if the executable is not found, assume in runs under docker with the port 27017 mapped from localhost to the docker container - and connect there.

Thinking about wording that might convey the right message: how about:
`Mongodb executable not found to start test MongoDB instance, trying to connect to docker mongodb` ?

Cheers,
Vlad